tests/test_filter_datasets.py

```diff
@@ -294,3 +341,109 @@ def test_stratified_crossval_splits_multiple_attributes(many_stimuli, crossval_f
+    if attribute_name == 'some_attribute' and attribute_value == 2 and invert_match is False:
+        inds = [2]
+        expected_fixations = fixations[inds]
+        compare_fixations(filtered_fixations, expected_fixations)
+
+
+@pytest.mark.parametrize('attribute_name , attribute_value, invert_match', [('some_attribute', 3, True)])
+def test_filter_fixations_by_attribute_some_attribute_invert_match(fixation_trains, attribute_name, attribute_value, invert_match):
+    fixations = fixation_trains[:]
+    filtered_fixations = filter_fixations_by_attribute(fixations, attribute_name, attribute_value, invert_match)
+    if attribute_name == 'some_attribute' and attribute_value == 3 and invert_match is False:
+        inds = list(range(0, 3)) + list(range(4, 8))
+        expected_fixations = fixations[inds]
+        compare_fixations(filtered_fixations, expected_fixations)
+
+
+@pytest.mark.parametrize('attribute_name , attribute_value, invert_match', [('task', 0, False)])
+def test_filter_scanpaths_by_attribute_task(fixation_trains, attribute_name, attribute_value, invert_match):
+    scanpaths = fixation_trains
+    filtered_scanpaths = filter_scanpaths_by_attribute(scanpaths, attribute_name, attribute_value, invert_match)
+    if attribute_name == 'task' and attribute_value == 0 and invert_match is False:
+        inds = [0, 2]
+        expected_scanpaths = scanpaths.filter_fixation_trains(inds)
+        compare_scanpaths(filtered_scanpaths, expected_scanpaths)
+
+
+@pytest.mark.parametrize('attribute_name , attribute_value, invert_match', [('multi_dim_attribute', [0, 3], False)])
+def test_filter_scanpaths_by_attribute_multi_dim_attribute(fixation_trains, attribute_name, attribute_value, invert_match):
+    scanpaths = fixation_trains
+    filtered_scanpaths = filter_scanpaths_by_attribute(scanpaths, attribute_name, attribute_value, invert_match)
+    if attribute_name == 'multi_dim_attribute' and attribute_value == [0, 3] and invert_match is False:
+        inds = [1]
+        expected_scanpaths = scanpaths.filter_fixation_trains(inds)
+        compare_scanpaths(filtered_scanpaths, expected_scanpaths)
+
+
+@pytest.mark.parametrize('attribute_name , attribute_value, invert_match', [('multi_dim_attribute', [0, 1], True)])
+def test_filter_scanpaths_by_attribute_multi_dim_attribute_invert_match(fixation_trains, attribute_name, attribute_value, invert_match):
+    scanpaths = fixation_trains
+    filtered_scanpaths = filter_scanpaths_by_attribute(scanpaths, attribute_name, attribute_value, invert_match)
+    if attribute_name == 'multi_dim_attribute' and attribute_value == [0, 1] and invert_match is True:
+        inds = [1, 2]
+        expected_scanpaths = scanpaths.filter_fixation_trains(inds)
+        compare_scanpaths(filtered_scanpaths, expected_scanpaths)
+
+
+@pytest.mark.parametrize('intervals', [([(1, 2), (2, 3)]), ([(2, 3), (3, 4)]), ([(2)]), ([(3)])])
```

Comment:
here the parameters and tests are sufficiently small, so here I see the point of parametrize.

Reply:
okay